R/breaks.r

```diff
@@ -59,7 +62,27 @@ log_breaks <- function(n = 5, base = 10) {
     if (max == min) return(base^min)
 
     by <- floor((max - min) / n) + 1
-    base^seq(min, max, by = by)
+    breaks <- base ^ seq(min, max, by = by)
+    relevant_breaks <- base ^ rng[1] <= breaks & breaks <= base ^ rng[2]
+    if (sum(relevant_breaks) >= (n - 2)) return(breaks)
+
+    steps <- 1
+    delta <- function(x) {
+      min(diff(log(sort(c(x, steps, base)), base = base)))
+    }
+    candidate <- 2:(base - 1)
+    while (length(candidate)) {
```

Comment:
I wonder if we should extract this out into a named function?

Reply:
Also would you mind adding a comment that explains the basic idea of the approach?